src/mono/mono/mini/interp/interp.c

```diff
@@ -6515,6 +6515,19 @@ MINT_IN_CASE(MINT_BRTRUE_I8_SP) ZEROP_SP(gint64, !=); MINT_IN_BREAK;
 			ip += 6;
 			MINT_IN_BREAK;
 		}
+		MINT_IN_CASE(MINT_STELEM_VT_NOREF) {
+			MonoArray *o = LOCAL_VAR (ip [1], MonoArray*);
+			NULL_CHECK (o);
+			guint32 aindex = LOCAL_VAR (ip [2], guint32);
+			if (aindex >= mono_array_length_internal (o))
+				THROW_EX (interp_get_exception_index_out_of_range (frame, ip), ip);
+
+			guint16 size = ip [5];
+			char *dst_addr = mono_array_addr_with_size_fast ((MonoArray *) o, size, aindex);
+			mono_gc_memmove_atomic (dst_addr, locals + ip [3], size);
```

Comment:
why not memcpy ? If we are not moving any refs that the GC might want to be stored atomically ?

Reply:
value_copy's fast path uses mono_gc_memmove_atomic. I do think it would make sense to use memcpy, so I can do that.